catalog/data_schema.go

```diff
@@ -0,0 +1,25 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+func DataSourceSchema() common.Resource {
+	return common.WorkspaceData(func(ctx context.Context, data *struct {
```

Comment:
are we standardising on `common.WorkspaceDataWithParams` or `common.WorkspaceData`?

Reply:
We discussed that we'll continue to use nested objects instead